homeassistant/components/proxmoxve/binary_sensor.py

```diff
@@ -96,8 +96,14 @@ def is_on(self):
         data = self.coordinator.data
 
         if data is None:
-            self.available = False
-            return False
+            self._state = None
+        else:
+            self._state = data["status"] == "running"
 
-        self.available = True
-        return self.coordinator.data["status"] == "running"
+        return self._state
```

Comment:
`self._state` could be removed, because it is only used by `is_on()`

```python
    def __init__(
        self,
        coordinator: DataUpdateCoordinator,
        unique_id,
        name,
        icon,
        host_name,
        node_name,
        vm_id,
    ):
        """Create the binary sensor for vms or containers."""
        super().__init__(
            coordinator, unique_id, name, icon, host_name, node_name, vm_id
        )

    @property
    def is_on(self):
        """Return the state of the binary sensor."""
        data = self.coordinator.data

        if data is None:
            return None
        else:
            return data["status"] == "running"
```